src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs

```diff
@@ -15,19 +14,23 @@ internal sealed class ILGeneratorImpl : ILGenerator
         private readonly MethodBuilder _methodBuilder;
         private readonly BlobBuilder _builder;
         private readonly InstructionEncoder _il;
+        private readonly ControlFlowBuilder _cfBuilder;
         private bool _hasDynamicStackAllocation;
         private int _maxStackSize;
         private int _currentStack;
         private List<LocalBuilder> _locals = new();
         private Dictionary<Label, LabelHandle> _labelTable = new(2);
         private List<KeyValuePair<MemberInfo, BlobWriter>> _memberReferences = new();
+        private List<ExceptionBlock> _exceptionStack = new();
+        private int _currentExStackIndex = -1;
```

Comment:
Should the `List<ExceptionBlock>` really be a `Stack<ExceptionBlock>`? It seems like that is precisely how it is being used.

Reply:
Good point, but seems I could not use `Stack<T>` from CoreLib